cachi2/core/package_managers/pip.py

```diff
@@ -1528,6 +1537,14 @@ def _process_pypi_req(
             )
         )
 
+    # hack
```

Comment:
I'm not fond of this, but I haven't been able to come up with a more elegant, Pythonic way of doing same...

Reply:
This section is better now, but maybe could be improved?

Reply:
I don't actually know the correct way to enable wheels, but you can set `allow_binary: bool = True` in  `input.py::PipPackageInput` (it's very close to the top of the class).

Then run a test scenario whose deps actually have wheels...

You could try cloning [StableCascade](https://github.com/Stability-AI/StableCascade) and using the [requirements.txt](https://gist.github.com/ben-alkov/e8540fe13a36b6fb9f195ffc99aedbd8) I generated for it - it has reqs with sdist-only, sdists+wheels, wheels-only, URL, **and** VCS, so it's a pretty good test. It does take a while to run :grin:

Then you can check the BOM...

Reply:
Thanks to Bruno for enlightening me about cachi2's CLI

```bash
cachi2 fetch-deps  '{"type": "pip", "allow_binary": true}' --output ~/temp/out/StableCascade --source ~/Source/StableCascade
```

Reply:
It's late and I'm not as familiar with this feature, so perhaps I'm missing something here, but is the de-duplication of the Component property for binaries the issue you are trying to solve?

If the property should be true if there is a wheel for any component being de-duplicated, can't we just update PropertySet.merge to do a logical OR instead of AND for the binary property?

Reply:
> can't we just update PropertySet.merge to do a logical OR instead of AND for the binary property

Now that I know what a PropertySet is, I can look into it :wink: